docker/ci/Dockerfile.nightly

```diff
@@ -0,0 +1,5 @@
+FROM rust:nightly
+RUN apt-get update && \
+    apt-get install -y protobuf-compiler && \
+    rustup component add rustfmt && \
+    cargo install cargo-audit
```

Comment:
I'd add an apt-get cache cleanup to be really neat about the image size.

Reply:
Also `cargo install` pulls loads of crates to compile, these caches should better be cleaned from the image as well.

We might want to build these images in CircleCI, where apt-get and cargo caches can be reused between runs like we do to speed up `cargo test` steps.